lib/api/contracts/model_contract.rb

```diff
@@ -51,6 +51,27 @@ def self.attribute(*attributes, &block)
       validate :readonly_attributes_unchanged
       validate :run_attribute_validations
 
+      # Performs validation of both, the contract and the underlying model and returns
+      # the merged validation errors for both of them.
+      def validate_contract_and_model
```

Comment:
But why do you want to inform the sender of the method call of the complexity you have in the method. The sender shouldn't care whether you have to also ask the model whether it is valid. The sender should be able to ask whether the contract is fulfilled. 

Why not call it `violations` then?